src/custom/pages/Swap/swapFlow/useSwapFlowContext.ts

```diff
@@ -0,0 +1,209 @@
+import { SwapFlowContext } from 'pages/Swap/swapFlow/types'
+import { useWeb3React } from '@web3-react/core'
+import { useSwapState } from 'state/swap/hooks'
+import { useDerivedSwapInfo, useDetectNativeToken } from 'state/swap/hooks'
+import { Field } from 'state/swap/actions'
+import usePriceImpact from 'hooks/usePriceImpact'
+import { useMemo } from 'react'
+import { WrapType } from 'hooks/useWrapCallback'
+import { GpEther as ETHER, WRAPPED_NATIVE_CURRENCY } from 'constants/tokens'
+import { useWalletInfo } from 'hooks/useWalletInfo'
+import { useCloseModals } from 'state/application/hooks'
+import { useAddPendingOrder } from 'state/orders/hooks'
+import { useAppData } from 'hooks/useAppData'
+import { useDispatch } from 'react-redux'
+import { AppDispatch } from 'state'
+import { SwapFlowAnalyticsContext } from 'pages/Swap/swapFlow/steps/analytics'
+import useENSAddress from 'hooks/useENSAddress'
+import { useSwapConfirmManager } from 'pages/Swap/hooks/useSwapConfirmManager'
+import { WrapEthInput } from 'pages/Swap/swapFlow/steps/wrapEthStep'
+import { useTransactionAdder } from 'state/enhancedTransactions/hooks'
+import { useWETHContract } from 'hooks/useContract'
+import { computeSlippageAdjustedAmounts } from 'utils/prices'
+import { PostOrderParams } from 'utils/trade'
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { OrderKind } from '@cowprotocol/contracts'
+import { NATIVE_CURRENCY_BUY_TOKEN } from 'constants/index'
+import { MAX_VALID_TO_EPOCH } from 'hooks/useSwapCallback'
+import { useUserTransactionTTL } from 'state/user/hooks'
+import { useGP2SettlementContract } from 'hooks/useContract'
+import { useUpdateAtom } from 'jotai/utils'
+import { addAppDataToUploadQueueAtom } from 'state/appData/atoms'
+
+const _computeInputAmountForSignature = (params: {
+  input: CurrencyAmount<Currency>
+  inputWithSlippage: CurrencyAmount<Currency>
+  fee?: CurrencyAmount<Currency>
+  kind: OrderKind
+}) => {
+  const { input, inputWithSlippage, fee, kind } = params
+  // When POSTing the order, we need to check inputAmount value depending on trade type
+  // If we don't have an applicable fee amt, return the input as is
+  if (!fee) return input
+
+  if (kind === OrderKind.SELL) {
+    // User SELLING? POST inputAmount as amount with fee applied
+    return input
+  } else {
+    // User BUYING? POST inputAmount as amount with no fee
+    return inputWithSlippage.subtract(fee)
+  }
+}
+
+function calculateValidTo(deadline: number): number {
+  // Need the timestamp in seconds
+  const now = Date.now() / 1000
+  // Must be an integer
+  const validTo = Math.floor(deadline + now)
+  // Should not be greater than what the contract supports
+  return Math.min(validTo, MAX_VALID_TO_EPOCH)
+}
+
+export function useSwapFlowContext(): SwapFlowContext | null {
```

Comment:
this is kind of a big hook, although cool that we get all the swap context.

Probably we will find ways later on to keep simplifying and cleaning up. Maybe when we refactor further for limit orders

Reply:
For sure, this hooks is an intermediate solution